device/tt_core_coordinates.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+ #include "device/tt_xy_pair.h"
+
+
+/*
+ * CoordSystem is an enum class that represents all types of coordinate
+ * systems that can be used to represent a core's location.
+ * This is used both for V1 and V2.
+ */
+enum class CoordSystem {
+    LOGICAL,
+    PHYSICAL,
+    VIRTUAL,
+    TRANSLATED,
+};
+
+// ************************************************************************************************
+// V1
+
+/*
+ * CoreType is an enum class that represents all types of cores
+ * present on the Tenstorrent chip.
+ */
+enum class CoreType {
+  ARC,
+  DRAM,
+  ETH,
+  PCIE,
+  TENSIX,
+  ROUTER_ONLY,
+};
+
+ struct CoreCoord_V1 : public tt_xy_pair {
```

Comment:
I prefer V1.  I was thinking we could use derived types to construct the base type: TensixCoreCoord() sets core_type to Tensix, then keep the CoreType internal only

Reply:
If I understand, `TensixCoreCoord` would me tt-metal abstraction? There would still be just CoreCoord_V1 in umd?

Reply:
yes.  is an API consideration for metal